### PR TITLE
Desktop: fix wording in settings

### PR DESCRIPTION
**Pull request by tessus**
Currently the wording in settings has been leading people to believe that the applications will be automatically updated.
In reality all that is done is checking for available updates.

Thus the new wording is: `Automatically check for updates`

~~For some reason this check was disabled on Linux, because during experiments with real auto-updates yielded negative results on Linux.
But since the current code only checks for updates, there is no reason to not allow this on Linux.~~

~~Laurent, as to the impact this change has on Linux (since I promised I will let you know): it allows people to enable the auto-check on Linux.~~

I have tested the change on macOS and Linux. According to the code there is no change that is Windows specific, so I am sure it has no impact on Windows.

### CODE REVIEW

**Comment by Daeraxa** [comment]
One small comment I have here is that this might be a bit confusing for some Linux users depending on how it is installed.
Now I know that the snap, flatpak etc. distributions are their own thing and I guess they could potentially manage something there and it cannot/should not account for every edge case, but the Joplinapp website recommends the linux install to be performed via the install and update script rather than just the plain appimage. 
The upgrade prompt will, I assume (but please correct me if wrong), prompt the user and take them straight to the .appimage download link (e.g. https://github.com/laurent22/joplin/releases/download/v2.5.12/Joplin-2.5.12.AppImage) which conflicts with the officially supported method by which the user installed the application.

Instead should the dialogue box presented to a Linux user also provide the the update script in a text box as well as the download link common to all platforms?

**Comment by tessus** [comment]
On one side I understand, but on the other side I have to say that this is not our concern. Please let me explain.

We have to make sure that the application works consistently across the board, or it least try to. We can't think of what might happen downstream. Otherwise you will never be able to innovate.

When it comes to packagers, they have to add patches to make the app behave on their respective distro. This is what the job of a packagers is.
When you look at rpm packages for Fedora (I'm taking this distro as an example, because I know the rpm repos and rpm sources best), they change default paths (for config files and logs), settings, add patches that are not yet in the latest release but in the source tree, and they even write patches which are not available uptream and mos tlikely will never be available upstream.

For this particular situation a packager has a few choices:

- rename the wording in the settings screen to reflect that it checks for updates upstream
- remove the setting and the check completely
- rewrite the check to look at the distro repo for updates instead

While the automatic check was disabled for Linux, the user could still check for updates manually. So this must have been even more confusing.

While I do understand your opinion, it is the job of the packager to handle these type of things. We have to make sure that someone who uses the official package has a proper way to check for updates.

**Comment by Daeraxa** [comment]
Sorry I don't think I was particularly clear but yes, that is exactly what I was trying to say that although there are other distribution channels they are not official for Joplin therefore it cannot be assumed that every feature will cater for them - basically that yes, it makes sense to not have to worry about them from the source - it is a downstream problem. Basically trying to acknowledge the argument but also dismissing it.

My main issue is simply usability from the Linux perspective for which I can empathise as a relatively new convert to daily driving Linux. There are definitely more and more people who aren't overly familiar with how Linux works
Essentially the main issue I see is from people who installed Joplin from the recommended script are unlikely to know where the .appimage even is (to be honest I had no idea either, it isn't something I've had to think about as I've only ever used the script which is far easier to deal with) which will end up with people having two "installed" versions yet their desktop shortcut will still only link to the scripted one.

This might be a non-issue and something which could be solved with a small amount of education but I think it helps to be unambiguous as opposed to recommending to use the script to handle all the hard stuff but then just throwing them an appimage and saying "do it yourself now". All I'm really thinking is that the popup on linux should have a section with the update script text as well as the download link.

**Comment by tessus** [comment]
I am not opposed to that idea. We could also just download the script instead.

To be fair. The web site does not say to use the script for updating, but for installing the app. But I get your point. We can also add info to the web site / documentation. Or download the script instead. But this is not my decision.

On the other hand. The check is so that people know that a new version is available. They don't have to download the image, but can run the script to update.

I believe that this change makes the experience more consistent. For me it makes no sense not to know when a new version is available. If people know how to run a script as a cronjob, they should also know how to replace a file. If not, there's a rather huge disconnect.

**Comment by Daeraxa** [comment]
I definitely agree, even without my suggestion I think the benefits significantly outweigh the negatives, just trying to make the arguments for things that I can see might pop up as a result

**Comment by laurent22** [comment]
I'm concerned that enabling auto-update is going to cause a lot of confusion on Linux (even if it's optional). People will enable it, then some AppImage in their system will be updated, but not the one they've actually installed (via some other package manager), so we'll get confused support posts about people "losing" their notes, or thinking they've upgraded the app when they haven't done anything.

Do Linux users really need this feature? Because I don't remember many people asking for it, or when they do usually the answer is to run the install script (which will also take care the icons, etc.). Given that there are other many ways to update in Linux, is there really a strong demand for this feature?

Or perhaps we should change the "Check for update" menu item to display a dialog "Please update using your package manager or the install script"?

**Comment by Daeraxa** [comment]
> I don't understand what that means. How is the AppImage updated, if you check for an update? There ia no auto-update.

I took it more that you will end up with a user who installed from flatpak or something will download the new .appimage, launch it and assume that it will update their Joplin instance which instead will just look like all their notes have vanished.

> Why are you so fixated on Linux? What about Windows and macOS? Same thing. 

Both only realistically have three distribution channels each which all have the same effect. "Store", "manual" (.exe/.dmg) and (far more rare) a package manager (chocolatey + brew). The fact is that 90% of Windows or MacOS users will be comfortable with those first two options, only people who really know what they are doing are likely going to be using the last option.

That simply isn't the same with Linux. It is easy to write things off and tell people to RTFM but the reality is that you have a lot of people currently migrating to "friendly" distros like Ubuntu, Linux Mint, pop!_os etc. where they are unfamiliar with the way things work.
I fell into the exact same trap with Joplin when I first installed on Linux. When you first install LM it makes a big deal on its introduction about using their own software manager which lists stuff from apt, snap and flathub. If you search for Joplin you can quite happily find it, install it and everything works just fine.
At this point most people don't even have a concept of what flatpak or snap even is, they just assume it is like the windows or app store. Then suddenly they are presented with this .appimage thing which also isnt the same as everything else (let alone .deb files). 
Basically it is very hard to directly correlate your experience of years of Windows or macOS usage with what Linux presents to you (and the differences between distro flavours is even more confusing) and whilst it is easy to dismiss this as user error or people not reading documentation the fact is that it can be very confusing indeed and of course the first place they come to for help will be the forums - not the package maintainers.

**Comment by hamishmb** [comment]
I think it would be nice if this said the version that was available, the current version, and then linked to the Linux installation part of the website - that way if the recommended installation method/script ever changes, the new information is always on the website - this avoids out of date information being confusing to those who haven't upgraded for ages.

What I do in my (far simpler) GUIs is have them download a plist (basically XML) that contains all the information to display eg new features when an update is available. The plist is then periodically updated when there's a new release, and the link never changes so the GUI never displays confusing outdated information.

As someone said before, downstream would probably just disable/remove the checker, like they do for eg Firefox. We could even provide some way of making this really easy for distro packagers or something. Tbh, I'd probably have used the Flatpak version if I'd realised it was a thing, unofficial as it is.

**Comment by laurent22** [comment]
I feel we should leave things as they are here. Changing it might solve some issues but also bring new ones. Anyway our supported way to install the app on Linux is via the shell script, and in that way it makes sense to remove the auto-update feature from the UI since it's not relevant.

**Comment by tessus** [comment]
I am confused and don't understand. What does this mean?

We leave the wording as wrong and the behavior as inconsistent as it is? Why? What's the logic behind it?

Because people could be too stupid and click on a button? Or download a file and double-click it in the Downloads folder and wonder why the update didn't work? I am asking because I don't understand the rationale. Users can still check for updates manually. And the same situation (which I still don't know what it is) you claim is the reason for not making the change does not apply when checking manually?

I truly don't understand. The only thing that would actually make any sense is to change the download link on Linux to the location of the script.

**Comment by laurent22** [comment]
I'm fine with the label update by the way. The thing is, there are two different changes in this pull request, one that's relatively uncontroversial (the label change), and another that's not (the default change). That's why we have this rule that there should be only one change per pull request otherwise we end up with a PR that never gets resolved.

**Comment by hamishmb** [comment]
I've never even been notified of updates on Linux.

Just had a look and as of 2.5.12 I don't see a checkbox relating to updates in the settings. I do have something about getting prereleases though, which is disabled.

**Comment by tessus** [comment]
@laurent22 

> The thing is, there are two different changes in this pull request, one that's relatively uncontroversial (the label change), and another that's not (the default change).

What default change?

The secod change is to allow Linux users to automatically check for updates, because they can already do that manually.
Greying out that setting is just as confusing as the wrong wording.

I added both to the same PR, because these 2 things often come up together in posts and questions. However, I can change this PR to only include the first change. It's not my problem when people are asking over and over again why the heck they can't check for updates on Linux. I certainly won't answer/reply anymore. Devil-may-care.

**Comment by laurent22** [comment]
Ok let's merge, thanks for the update.
